docs/design/proposals/encryped-with-keyprotect.md

```diff
@@ -0,0 +1,67 @@
+# Encrypted volumes with IBM HPCS or Key Protect
+
+IBM Cloud™ Hyper Protect Crypto Services is a key management and cloud hardware
+security module (HSM). It is designed to enable a user to take control of their
+cloud data encryption keys and cloud hardware security models. To support this
+KMS integration in Ceph CSI and thus enable the HPCS users to make use of the
+same in RBD volume encrypted operations, below things are considered.
+
+## Connection to IBM HPCS/key Protect service
+
+Below parameters/values can be used to establish the connection to the HPCS
+service from the CSI driver and to make use of the encryption operations:
+
+```console
+* KMS_SERVICE_NAME=[kms_service_name]
+A unique name for the key management service within the project.
+
+* SERVICE_INSTANCE_ID=[service_instance_id]
+The Instance ID of the IBM HPCS service, ex:  crn:v1:bluemix:public:hs-crypto:us-south:a/5d19cf8b82874c2dab37e397426fbc42:e2ae65ff-954b-453f-b0d7-fc5064c203ce::
+
+* SERVICE_API_KEY=[service_api_key]
+Ex:  06x6DbTkVQ-qCRmq9cK-p9xOQpU2UwJMcdjnIDdr0g2R
+
+* CUSTOMER_ROOT_KEY=[customer_root_key]
+Ex: c7a9aa91-5cb5-48da-a821-e85c27b99d92
+
+* KP_REGION = [region of the key protect service]
+Ex: us-south-2
```

Comment:
do we have any link on how to retrieve these pieces of information? if yes good to add it.